### DISCUSSION TITLE
Install Niri for MOST DISTROS (script)

### DISCUSSION THREAD

**Discussion by amatsagu** [YaLTeR/niri#392] [Ideas]
Hey, I like this project so far, I've seen that at least for now it's available Gentoo, NixOS, Arch and its variants but it rather skips those more casual friendly distros like Debian, Ubuntu, Mint or base Fedora. I would like to address this - what do you think about maybe releasing an "official" installation (bash) script that would just download latest release binary files and neatly organize them on user computer or make it download latest version of rust (stable), then compile own latest stable release of Niri and copy files to correct places. It would be then something very similar to what Deno or Rust install scripts does and should realistically work for all distros except those extreme ones like Gentoo, Linux from Scratch or NixOs but dozens of other would be covered.

I'm also down for creating such installation script & maintaining it if you like the idea. Users could just paste:
```sh
curl --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/YaLTeR/niri/main/install.sh | sh
```
*(example command, but it would be something very similar)*
*("special" distros like NixOS or Gentoo would still need its own dedicated package, about updating, how about having command like `niri update` that would just check if there's new release and if it is - rerun install script)*

**Comment by amatsagu** [comment]
I forgot to mention - higher idea is an attempt to cover a bunch of different distros but it should be seen as optional, kinda last resort installation option that would be listed at the whole bottom (kinda like some apps offer flatpack as alternative). It would still be recommended to install Niri as system package if your package manager is supported.

**Comment by YaLTeR** [comment]
Hey! The problems are that compositors link to a lot of system dependencies (Ubuntu 22.04 doesn't have a new enough pipewire for instance) and that they need to install various systemd and desktop files in the right locations that may vary from distro to distro. I don't really want to deal with all this in the main repo.

For Fedora I've got a COPR linked in the getting started page; for Ubuntu someone could make a PPA which I could link. I think that's a better approach here. I don't really mind that there isn't an easier official installation method for all distros—they'll pick it up eventually if users show interest.

**Reply by amatsagu** [reply]
Ok, yeah - I think that's fine. I tried it on Debian and had no issue with pipewire at all but not sure about Ubuntu. Only issue I had is that some apps launch with weird / cursed bars but I assume you can just disable them in niri config (I still use default settings for time being). I have no interest in dealing with PPA for now, those are far more annoying than just using installation script.
